content/courses/advanced-ada/chapters/subprograms.rst

```diff
@@ -716,10 +1000,161 @@ makes the overload resolution algorithm reasonably efficient.
 Operator Overloading
 --------------------
 
-.. todo::
+We've seen :ref:`previously <Adv_Ada_Operators>` that we can define custom
+operators for any type. We've also seen that subprograms can be
+:ref:`overloaded <Adv_Ada_Overloading>`. Since operators are functions, we're
+essentially talking about operator overloading, as we're defining the same
+operator (say :ada:`+` or :ada:`-`) for different types.
 
-    Complete section!
+As another example of operator overloading, in the Ada standard library,
+operators are defined for the :ada:`Complex` type of the
+:ada:`Ada.Numerics.Generic_Complex_Types` package. This package contains not
+only the definition of the :ada:`+` operator for two objects of :ada:`Complex`
+type, but also for combination of :ada:`Complex` and other types. For instance,
+we can find these declarations:
+
+.. code-block:: ada
+
+    function "+" (Left, Right : Complex) return Complex;
+    function "+" (Left : Complex;   Right : Real'Base) return Complex;
+    function "+" (Left : Real'Base; Right : Complex)   return Complex;
+
+This example shows that the :ada:`+` operator |mdash| as well as other
+operators |mdash| are being overloaded in the :ada:`Generic_Complex_Types`
+package.
+
+.. admonition:: In the Ada Reference Manual
+
+    - `6.6 Overloading of Operators <http://www.ada-auth.org/standards/12rm/html/RM-6-6.html>`_
+    - `G.1.1 Complex Types <http://www.ada-auth.org/standards/12rm/html/RM-G-1-1.html>`_
+
+Operator Overriding
+-------------------
+
+We can also override operators of derived types. This allows for modifying the
+behavior of operators for certain types.
```

Comment:
Maybe "operators for the corresponding derived types." instead of "certain types"?